sdk/helper/keysutil/policy.go

```diff
@@ -2067,3 +2062,163 @@ func (p *Policy) EncryptWithFactory(ver int, context []byte, nonce []byte, value
 
 	return encoded, nil
 }
+
+func (p *Policy) KeyVersionCanBeUpdated(keyVersion int, isPrivateKey bool) error {
+	keyEntry, err := p.safeGetKeyEntry(keyVersion)
+	if err != nil {
+		return err
+	}
+
+	if !p.Type.ImportPublicKeySupported() {
+		return errors.New("provided type does not support importing key versions")
+	}
+
+	isPrivateKeyMissing := keyEntry.IsPrivateKeyMissing()
+	if isPrivateKeyMissing && !isPrivateKey {
+		return errors.New("cannot add a public key to a key version that already has a public key set")
+	}
+
+	if !isPrivateKeyMissing {
+		return errors.New("private key imported, key version cannot be updated")
+	}
+
+	return nil
+}
+
+func (p *Policy) ImportPrivateKeyForVersion(ctx context.Context, storage logical.Storage, keyVersion int, key []byte) error {
+	keyEntry, err := p.safeGetKeyEntry(keyVersion)
+	if err != nil {
+		return err
+	}
+
+	// Parse key
+	parsedPrivateKey, err := x509.ParsePKCS8PrivateKey(key)
```

Comment:
As a nit that we can address later: this doesn't use the same error handling as above around Ed25519 and RSA-PSS keys, meaning this will fail on certain types of imports. We can address this after this change merges. :-)